Sources/SwiftGD/SwiftGD.swift

```diff
@@ -49,6 +49,26 @@ public class Image {
 		}
 	}
 
+    public convenience init?(data: Data) {
+
+        // Bytes must not exceed int32 as limit by `gdImageCreate..()`
+        guard data.count < Int32.max else { return nil }
+
+        // Creates a gdImage pointer if given data represents an image in either PNG or JPEG raster format
+        let createImage: (UnsafeMutablePointer<UInt8>) -> gdImagePtr? = { pointer in
+            let size = Int32(data.count)
+            let rawPointer = UnsafeMutableRawPointer(pointer)
+            return gdImageCreateFromPngPtr(size, rawPointer) // Returns if valid PNG image data
+                ?? gdImageCreateFromJpegPtr(size, rawPointer) // Returns if valid JPEG image data
```

Comment:
Is `gdImageCreateFromGifPtr()` also available? We might as well go the whole hog.